modules/ardrivo/test/Arduino.cxx

```diff
@@ -0,0 +1,268 @@
+    num = constrain(11, 2, 10);
+    REQUIRE(num == 10);
+}
+
+TEST_CASE("Re-maps number from one range to another" "[map]") {
+    long num = 1023;
+    long num2 = map(num, 0, 1023, 0, 255);
+    REQUIRE(num2 == 255);
+}
+
+TEST_CASE("Check is char i letter", "[isAlpha]") {
+    REQUIRE(isAlpha('a') == true);
+    REQUIRE(isAlpha('1') == false);
+    ;
+}
+
+TEST_CASE("Check if a char is alphanumeric", "[isAlphaNumeric]") {
+    REQUIRE(isAlphaNumeric('a') == true);
+    REQUIRE(isAlphaNumeric('1') == true);
+    REQUIRE(isAlphaNumeric('!') == false);
+}
+
+TEST_CASE("Is the character a ascii", "[isAscii]") {
+    // REQUIRE(isAscii('a') == 97); //Eeehhh??????????????
+
+    int var = 97;
+    //std::cout << (char)var;
+}
+
+TEST_CASE("Check if the character a control character", "[isControl") {
+    REQUIRE(isControl('\t') == true);
+    REQUIRE(isControl('\n') == true);
+    REQUIRE(isControl('\v') == true);
+}
+TEST_CASE("Checks if the charahcter a digit" "[isDigit]") {
+    REQUIRE(isDigit('1') == true);
+    REQUIRE(isDigit('a') == false);
+    REQUIRE(isDigit('#') == false);
+}
+
+TEST_CASE("Checks if a character is a graphical represention" "[isGraph]") {
+    REQUIRE(isGraph('a') == true);
+    REQUIRE(isGraph('1') == true);
+    REQUIRE(isGraph(' ') == false);
+}
+
+TEST_CASE("Checks if a character is hexadecimal", "[isHexadecimalDigit]") {
+    REQUIRE(isHexadecimalDigit('a') == true);
+    REQUIRE(isHexadecimalDigit('!') == false);
+}
+
+TEST_CASE("Checks if a character is lowercase", "[isLowerCase") {
+    REQUIRE(isLowerCase('a') == true);
+    REQUIRE(isLowerCase('A') == false);
+}
+
+TEST_CASE("Checks if a character is printable", "[isPrintable]") {
+    REQUIRE(isPrintable('a') == true);
+    REQUIRE(isPrintable('1') == true);
+    REQUIRE(isPrintable(' ') == true);
+}
+
+TEST_CASE("Checks if a character is a punctuation" "[isPunct") {
+    REQUIRE(isPunct('a') == false);
+    REQUIRE(isPunct(',') == true);
+    REQUIRE(isPunct('?') == true);
+}
+
+TEST_CASE("Checks if a character contains white-space", "[isSpace]") {
+    REQUIRE(isSpace('a') == false);
+    REQUIRE(isSpace('\t') == true);
+    REQUIRE(isSpace(' ') == true);
+}
+
+TEST_CASE("Checks if a character is uppercase" "[isUpperCase") 
+{
+    REQUIRE(isUpperCase('A') == true);
+    REQUIRE(isUpperCase('a') == false);
+}
+
+TEST_CASE("Checks if a character is a blank character" "[isWhiteSpace]") 
+{
+    REQUIRE(isWhitespace(' ') == true);
+    REQUIRE(isWhitespace('\t') == true);
+    REQUIRE(isWhitespace('a') == false);
+}
+
+TEST_CASE("Checks if it returns a random number" "[isRandom]") 
+{
+    auto i = GENERATE(take(100, filter([](int i) { return i; }, random(-100, 100))));
+    REQUIRE(i < 100);
+    REQUIRE(i > -100);
+}
+TEST_CASE("Test randomSeed", "[randomSeed]") 
+{
+    std::srand(2);
+    int num = std::rand();
+    int num2 = std::rand();
+
+    std::srand(2);
+    int num3 = std::rand();
+
+    REQUIRE(num != num2);
+    REQUIRE(num == num3);
+}
+
```

Comment:
The test is lying, since it is testing C random function rather than the Arduino ones